jax/_src/compiler.py

```diff
@@ -53,6 +53,15 @@
     ),
 )
 
+_DISABLE_COMPILER_REMAT_OPTIMIZATION_PASS = config.bool_flag(
+    "jax_compiler_enable_remat_pass",
+    config.bool_env('JAX_COMPILER_ENABLE_REMAT_PASS', True),
```

Comment:
It is simpler to understand if all is implemented as enabling that pass.
So maybe _DISABLE_COMPILER_REMAT_OPTIMIZATION_PASS -> _ENABLE_COMPILER_REMAT_OPTIMIZATION_PASS
And set the default to False? If people want to optimize it, they set it to True?